File: test-plugins/org.yakindu.sct.generator.cpp.test/gtests/OutEventLifeCycleTest/OutEventLifeCycleTest.cc
/**
* Copyright (c) 2017 committers of YAKINDU and others.
* All rights reserved. This program and the accompanying materials
* are made available under the terms of the Eclipse Public License v1.0
* which accompanies this distribution, and is available at
* http://www.eclipse.org/legal/epl-v10.html
*
* Contributors:
*     committers of YAKINDU - initial API and implementation
*/
#include <string>
#include "gtest/gtest.h"
#include "OutEventLifeCycle.h"
#include "sc_types.h"
OutEventLifeCycle* statechart;

class StatemachineTest : public ::testing::Test{
	protected:
	virtual void SetUp() {
		statechart = new OutEventLifeCycle();
		statechart->init();
	}
	virtual void TearDown() {
		delete statechart;
	}
};

void init(bool sndCycle){
	statechart->enter();
	statechart->raise_e();
	statechart->runCycle();
	if (sndCycle) {
		statechart->runCycle();
	}
}

TEST_F(StatemachineTest, availableAfterCycle) {
	
	
	statechart->enter();
	
	statechart->raise_e();
	
	statechart->runCycle();
	
	EXPECT_TRUE(statechart->getDefaultSCI()->isRaised_f());
	
}
TEST_F(StatemachineTest, availableWithinCycle) {
	
	
	init(false);
	
	EXPECT_TRUE(statechart->getDefaultSCI()->get_f_available_in_cycle());
	
}
TEST_F(StatemachineTest, unvailableWithin2ndCycle) {
	
	
	init(true);
	
	EXPECT_TRUE(!statechart->getDefaultSCI()->get_f_available_in_next_cycle());
	
}
TEST_F(StatemachineTest, unvailableAfter2ndCycle) {
	
	
	init(true);
	
	EXPECT_TRUE(!statechart->getDefaultSCI()->isRaised_f());
	
}

File: test-plugins/org.yakindu.sct.generator.java.test/test-gen/org/yakindu/sct/generator/java/test/OutEventLifeCycleTest.java
/**
* Copyright (c) 2017 committers of YAKINDU and others.
* All rights reserved. This program and the accompanying materials
* are made available under the terms of the Eclipse Public License v1.0
* which accompanies this distribution, and is available at
* http://www.eclipse.org/legal/epl-v10.html
*
* Contributors:
*     committers of YAKINDU - initial API and implementation
*/

package org.yakindu.sct.generator.java.test;

import org.junit.*;
import static org.junit.Assert.*;
import org.yakindu.scr.outeventlifecycle.OutEventLifeCycleStatemachine;
import org.yakindu.scr.outeventlifecycle.OutEventLifeCycleStatemachine.State;

/**
 * Unit TestCase for OutEventLifeCycle
 */
@SuppressWarnings("all")
public class OutEventLifeCycleTest {
	private OutEventLifeCycleStatemachine statemachine;	
	
	@Before
	public void setUp() {
		statemachine = new OutEventLifeCycleStatemachine();
		statemachine.init();
	}

	@After
	public void tearDown() {
		statemachine = null;
	}
	
	@Test
	public void testAvailableAfterCycle() {
		statemachine.enter();
		statemachine.raiseE();
		statemachine.runCycle();
		assertTrue(statemachine.isRaisedF());
	}
	@Test
	public void testAvailableWithinCycle() {
		init(false);
		assertTrue(statemachine.getF_available_in_cycle());
	}
	@Test
	public void testUnvailableWithin2ndCycle() {
		init(true);
		assertTrue(!statemachine.getF_available_in_next_cycle());
	}
	@Test
	public void testUnvailableAfter2ndCycle() {
		init(true);
		assertTrue(!statemachine.isRaisedF());
	}
	public void init(boolean sndCycle) {
		statemachine.enter();
		statemachine.raiseE();
		statemachine.runCycle();
		if (sndCycle) {
			statemachine.runCycle();
		}
	}
}
